extensions/chghost-3.2.md

```diff
@@ -5,59 +5,87 @@ copyrights:
+the same channel as the target client or have the client on a `MONITOR` list.
+Servers SHOULD additionally send the `CHGHOST` message to the client whose
+user or host has changed if the client supports the `chghost` capability.
 
-The CHGHOST message is one of the following:
+When the server sends a `CHGHOST` message to a client, it MUST defer doing so
+until both successful `NICK` and `USER` commands have been received by the
+server.  The server MAY choose to defer it until after registration is
+completed, for example if a valid SASL authentication during client
+registration triggers an assignment of a virtual host. If the server does not
+defer it until registration is completed, and either the user or the host of
+the client changes, the server MUST send a new `CHGHOST` message.
 
-    :nick!user@host CHGHOST user new.host.goes.here
+## The `CHGHOST` message
 
-This message represents that the user identified by nick!user@host has changed
-host to another value. The first parameter is the user of the client. The
-second parameter is the new host the client is using.
+The `CHGHOST` message is as follows:
 
-On irc daemons with support for changing the user portion of a client, the
-second form may appear:
+    :nick!old_user@old_host.local CHGHOST new_user new_host.local
 
-    :nick!user@host CHGHOST newuser host
+The `new_user` parameter represents the user's "username" or "ident" which may
+or may not have changed in the CHGHOST process.
 
-If specified, a client may also have their user and host changed at the same
-time:
+The `new_host.local` parameter represents the new hostname for the user which
+may or may not have changed in the CHGHOST process.
 
-    :nick!user@host CHGHOST newuser new.host.goes.here
+Servers that implement ident checking might choose to prefix a username with a tilde `~` character to indicate missing confirmation from an ident server. This MUST be considered part of the username and included in any CHGHOST messages where relevant.
 
-This second and third form should only be seen on IRC daemons that support
-changing the user field of a user.
+    :nick!~old_user@old_host.local CHGHOST ~new_user new_host.local
 
-In order to take full advantage of the CHGHOST message, clients must be modified
-to support it. The proper way to do so is this:
+## Fallback when the `chghost` capability has not been negotiated
 
-1) Enable the chghost capability at capability negotiation time during the
-   login handshake.
+When the capability is not enabled for other clients who share channels with or
+monitor the changed client, servers SHOULD send fallback messages to simulate the client
+reconnecting. This allows clients to keep their user state up to date. For
+shared channels, the simulated events SHOULD include appropriate `QUIT`, `JOIN`
+and `MODE` commands, to restore membership and user channel modes. For monitored
+clients, the events SHOULD include appropriate `RPL_MONOFFLINE` and
+`RPL_MONONLINE` numerics.
```

Comment:
This seems questionable since it'll pointlessly highlight people.

Reply:
Well, highlighting depends on the client. This allows clients that are monitoring users but aren't in shared channels to have an up to date mask, which is a requirement for various things.

Reply:
Can't find much discussion of it other than these two comments:

https://github.com/ircv3/ircv3-specifications/pull/364#issuecomment-464385022
https://github.com/ircv3/ircv3-specifications/pull/364#pullrequestreview-204541454

Not sure which implementations if any do this. As a SHOULD it's not a breaking change, but it should definitely be errataed if kept, as you noted elsewhere.

Reply:
From a glance at inspircd, unreal, charybdis, hybrid, none of these servers send monitor numerics in the fallback case, nor send the cap-enabled message to monitoring clients.

Given we aren't routinely extending MONITOR to support other forms of change notification (e.g. SETNAME) there's probably an argument for dropping it from this.

There's probably also an argument for extending MONITOR more thoroughly, but given lack of implementation, I reckon we can defer that.

Thoughts @SadieCat @syzop @slingamn?